### ISSUE TITLE
GestureRecognizers inhibit CollectionView Selection on Android

### ISSUE THREAD

**Issue by david-maw** [dotnet/maui#6644]
### Description

In a CollectionView I use click to select an item and have a gesturerecognizer attached so as to be able to do other things in response to the click. A Windows build selects the item then calls the GestureRecognizer, on Android it does not select the item before calling the GestureRecognizer. 

### Steps to Reproduce

1. Unzip the project from
[MAUIListEntry-TapGestureRecognizer.zip](https://github.com/dotnet/maui/files/8588249/MAUIListEntry-TapGestureRecognizer.zip)
2. Cmpile and run it on Windows, click each item in the list in turn, you should see this:
![MauiListEntry-TapGestureRecognizer-Windows](https://user-images.githubusercontent.com/12423046/165888105-c5590b80-364d-4185-b58c-9259c2f630c1.gif)
3. repeat on Android, this time clicking the items will do nothing and you'll just see this:
![MauiListEntry-TapGestureRecognizer-Android](https://user-images.githubusercontent.com/12423046/165888170-242d8e0b-9fc2-4e31-ae7f-0f7cd8075995.gif)
 
Also, the default line spacing seems to be wildly different between Windows and Android.

### Version with bug

Release Candidate 2 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

Android, I was *not* able test on other platforms

### Affected platform versions

Android 11

### Did you find any workaround?

No

**Comment by kristinx0211** [comment]
verified repro on android with above project, vs 17.3.0 Preview 1.0 [32426.537.main].

**Comment by kristinx0211** [comment]
still an issue on android with 17.3.0 Preview 2.0 [32507.29.main].

**Comment by StephenWin** [comment]
I hope this will get fixed soon, it has been over a year waiting and it is very important to use the CollectionView on Android.  I want selection on single tap and navigation on 2 taps.  This works easy on Windows but as the Android ignores selection when the GestureRecognizers are used for double tap it is useless.

**Comment by EvoZ1310** [comment]
I have the same issue on MacCatalyst as well

**Comment by axa88** [comment]
Verified this on with VS Version 17.8.5

Its been years... 
[One bug](https://stackoverflow.com/questions/74823748) prevents me from using Border so i switch to the suggested Layout.
Then [the next bug](https://github.com/dotnet/maui/issues/9567) states i must use Border as the `CollectionView` wont generate a `SelectedItem` otherwise.
So in the stalemate i switch to a `TapGestureRecognizer` to substitute for `SelectedItem` only to find this problem and i cant use it either.
Sometimes im shocked `Maui` is a released product.

**Comment by wesleykuhn** [comment]
@axa88 So sadly for you mate 😞 
But you can do a workaround that aways works for me. Instead of use the selection command of the CollectionView, place a gesture recognizer on the first view of the item template, then you can send the binding object itself as command paramenter. Then you don't a selected item property on your VM to watch the selected item on your CollectionView. It should be something like this:

```
<CollectionView
    ...
    SelectionMode="None">
    
    <CollectionView.ItemTemplate>
        <DataTemplate x:DataType="MyModelClass">
            <Frame>
                <Frame.GestureRecognizers>
                    <TapGestureRecognizer Command="{Binding Source={RelativeSource AncestorType={x:Type ContentPage}}, Path=BindingContext.ItemSelectedCommand}" CommandParameter="{Binding .}" />
                </Frame.GestureRecognizers>

                <Some components of the item layout/>
            </Frame>
        </DataTemplate>
    </CollectionView.ItemTemplate>
</CollectionView>
```

Basically the ideia is **access the binding context of the entire ContentPage (via AncestorType) on the CollectionView item and call the command from there**. That's not a very good code, but it's what is aways working for me.
And don't forget to accept the model which is used on the template of CollectionView as parameter on the command:

```
private ICommand itemSelectedCommand;
public ICommand ItemSelectedCommand ??= new Command<MyModelClass>((arg) => ItemSelectedCommandExecute(arg));

private void ItemSelectedCommandExecute(MyModelClass selected)
{
    ...
}
```

Hope that helps you.

**Comment by axa88** [comment]
@wesleykuhn 
Does this work with a `DataTemplateSelector` and use of modern `Border` rather than `Frame`?

**Comment by wesleykuhn** [comment]
> @wesleykuhn Does this work with a `DataTemplateSelector` and use of modern `Border` rather than `Frame`?

I never did it with DataTemplateSelector, maybe it will work, because it will seek for the parent of the item template view, using its type as criteria. Border has gesture recognizer, you can use it.

**Comment by ghuotvez** [comment]
Still having this issue under Android, but works fine in Win.

**Comment by BohdanRomaniuk** [comment]
Unfortunately this is still an issue as of this day, which reproduced for me on both Android & iOS platforms. 
Using MAUI 8.0.100
Here are the possible workaround I have found for an Item template

```
public partial class CustomContentView : ContentView
{
    public CustomContentView()
    {
        InitializeComponent();
        GestureRecognizers.Add(new TapGestureRecognizer() { Command = new Command(OnTapped) });
    }

    protected virtual void OnTapped()
    {
        if (Parent is CollectionView cv)
        {
            cv.SelectedItem = BindingContext;
        }
    }
}
```

**Comment by LogishaSelvarajSF4525** [comment]
This issue has been verified in Visual Studio Code 1.103.0 with MAUI versions 9.0.90. It can be reproduced on the Android platform.

**Comment by LaoR-UA** [comment]
Still an issue with MAUI version 10.0.2
